homeassistant/components/moehlenhoff_alpha2/__init__.py

```diff
@@ -32,6 +37,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     hass.data.setdefault(DOMAIN, {})
     hass.data[DOMAIN][entry.entry_id] = coordinator
 
+    async def async_handle_set_datetime_service(call: ServiceCall):
+        str_value: Any = call.data.get("datetime")
+        value = datetime.fromisoformat(str_value)
+
+        await base.set_datetime(value)
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_SET_DATETIME,
+        async_handle_set_datetime_service,
+        SERVICE_SCHEMA_SET_DATETIME,
+    )
```

Comment:
Why not create a button entity that syncs the current time (instead of this service)?

Reply:
That would most certainly also be an appropriate solution. I was considering how to best make this available and thought that a service would be the most flexible approach, even though I'm not sure what use case people could come up with, other than syncing with the current time. I simply did not want to limit the possibilities this could provide. 

This said, if a button entity is preferred, I see no issue in moving to such an implementation.

Reply:
> This said, if a button entity is preferred, I see no issue in moving to such an implementation.

I think it would be a better experience for the users. A simple button that shows up in the interface and could also be automated if wanted.